python/tvm/relay/testing/temp_op_attr.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+""" Defines a TempOpAttr class that allows temporarily changing an attr of the
+operator to allow unit testing. This is useful for AlterOpLayout and Legalize
+tests."""
+
+from tvm import relay
+
+class TempOpAttr(object):
+    """ Temporarily changes the attr of an op. """
+    def __init__(self, op_name, attr_key):
```

Comment:
document all the arguments https://docs.tvm.ai/contribute/document.html#document-python

```python
"""
Parameters
----------

Examples
----------
"""
```

Reply:
Can be combine the set and reset together?

```python
# set op attrkey to value
with TempOpAttr(op_name, attr_key, attr_value):
```

Reply:
Good idea! Will update soon.